### ISSUE TITLE
Replace `kernel::str::CStr` by `core::ffi::CStr`

### ISSUE THREAD

**Issue by BennoLossin** [Rust-for-Linux/linux#1075]
`std::ffi::CStr` was moved to `core::ffi::CStr` in rust 1.64, making it available to the kernel. We have our own version of `CStr` in `kernel::str`, since the standard library one wasn't available. Rust 1.77 also added C-string literals `c"abc"` that automatically terminate strings with `\0`. The kernel has a `c_str!` macro for this purpose. We should use the types from the standard library and remove our own versions. The things to do would be
- change uses of `kernel::str::CStr` to `core::ffi::CStr` and also change `c_str!("...")` into `c"..."`
- remove `kernel::str::CStr` and the macro

---
This requires submitting a proper patch to the LKML and the Rust for Linux mailing list. Please recall to test your changes, to use a proper title for the commit, to sign your commit under the Developer's Certificate of Origin and so on. Please see https://docs.kernel.org/process/submitting-patches.html and https://rust-for-linux.com/contributing for details.

Please take this issue only if you are new to the kernel development process and you would like to use it as a test to submit your first patch to the kernel. Please do not take it if you do not plan to make other contributions to the kernel.

**Comment by vincenzopalazzo** [comment]
I will look into it 😸

**Comment by BennoLossin** [comment]
We try to give good-first-issues to people who do not yet have commits in the kernel. It helps folks getting familiar with the kernel development process. We also try to pick easy/trivial fixes as good-first-issues. AFAIK you already submitted a good-first-issue, so it would be best to leave this up to someone else :)

**Comment by vincenzopalazzo** [comment]
Make sense!

**Comment by vadorovsky** [comment]
I can take it.

**Comment by Steffo99** [comment]
Hi!

I'd be interested in taking this issue to both learn to contribute to Rust for Linux and to submit my first patch to the Linux kernel, which is a requirement to pass the Kernel Hacking university course I've been attending.

@vadorovsky, are you still working on it?  
(I'm sending you a ping on the Fediverse too as soon as I post this, just to make sure that you see it...)

Can I take this issue?

**Comment by Steffo99** [comment]
Got an answer by @vadorovsky on the Fediverse!

Seems like he's still working on it :)

I'll look for another issue!

**Comment by vadorovsky** [comment]
Yes, I'm still working on it. Here is my WIP branch: https://gitlab.com/vadorovsky/linux/-/commits/rust-cstr

Before sending it, I still want to polish it a bit. I'm also thinking how to handle the fact that `alloc::CString` doesn't have any equivalent of `try_from_fmt`, but my idea for now is to add `CStringExt`. I'm coding it right now. If I'm happy from the result, I will add it and send it to the ML.

**Comment by vadorovsky** [comment]
Sent a patch, only with `CStr` removal:

https://lore.kernel.org/rust-for-linux/20240714160238.238708-1-vadorovsky@gmail.com/T/#u

**Comment by tamird** [comment]
Could someone assign this to me? I've sent https://lore.kernel.org/all/20250203-cstr-core-v8-0-cb3f26e78686@gmail.com/t/#u with @vadorovsky's permission.

**Comment by brambhattabhishek** [comment]
Could some one assign to me  this ?

**Comment by tamird** [comment]
I have already done this work, the patches are pending review.

**Comment by ojeda** [comment]
With the update to the docs of `c_str!` applied now, this is essentially done, unless I am missing something, so closing.
